virtualhostkeys.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package description
+
+import (
+	"encoding/base64"
+
+	"github.com/juju/errors"
+	"github.com/juju/schema"
+)
+
+type virtualHostKeys struct {
+	Version         int               `yaml:"version"`
+	VirtualHostKeys []*virtualHostKey `yaml:"virtual-host-keys"`
+}
+
+type virtualHostKey struct {
+	ID_      string `yaml:"id"`
+	HostKey_ string `yaml:"host-key"`
+}
+
+// ID implements VirtualHostKey
+func (f *virtualHostKey) ID() string {
+	return f.ID_
+}
+
+// HostKey implements VirtualHostKey
+func (f *virtualHostKey) HostKey() []byte {
+	// Here we are explicitly throwing away any decode error. We encode
+	// a byte array, so we know that the stored host key can be decoded.
```

Comment:
Can we just add a note that the decodability of the content is validated during import etc